programs/local/LocalServer.cpp

```diff
@@ -544,6 +544,17 @@ void LocalServer::processConfig()
     if (mark_cache_size)
         global_context->setMarkCache(mark_cache_size);
 
+    /// Size of cache for uncompressed blocks of MergeTree indices. Zero means disabled.
+    size_t index_uncompressed_cache_size = config().getUInt64("index_uncompressed_cache_size", 0);
+    if (index_uncompressed_cache_size)
+        global_context->setIndexUncompressedCache(index_uncompressed_cache_size);
+
+    /// Size of cache for index marks (index of MergeTree skip indices). It is necessary.
+    /// Specify default value for index_mark_cache_size explicitly!
+    size_t index_mark_cache_size = config().getUInt64("index_mark_cache_size", 0);
```

Comment:
This major performance degradation went unnoticed for two years.

Reply:
IIRC, before this PR, there is no mark/uncompressed cache for secondary indices at all. I don't think it's a degradation.